epochStart/metachain/rewards.go

```diff
@@ -141,20 +171,49 @@ func (rc *rewardsCreator) CreateRewardsMiniBlocks(metaBlock *block.MetaBlock, va
 		miniBlocks[shardId].TxHashes = append(miniBlocks[shardId].TxHashes, rwdTxHash)
 	}
 
-	for shId := uint32(0); shId < rc.shardCoordinator.NumberOfShards(); shId++ {
-		sort.Slice(miniBlocks[shId].TxHashes, func(i, j int) bool {
-			return bytes.Compare(miniBlocks[shId].TxHashes[i], miniBlocks[shId].TxHashes[j]) < 0
-		})
+	return nil
+}
+
+func (rc *rewardsCreator) addCommunityRewardToMiniBlocks(
+	miniBlocks block.MiniBlockSlice,
+	epochStartMetablock *block.MetaBlock,
+) error {
+	rwdTx, rwdTxHash, shardId, err := rc.createCommunityRewardTransaction(epochStartMetablock)
+	if err != nil {
+		return err
 	}
+	rc.currTxs.AddTx(rwdTxHash, rwdTx)
+	miniBlocks[shardId].TxHashes = append(miniBlocks[shardId].TxHashes, rwdTxHash)
 
-	finalMiniBlocks := make(block.MiniBlockSlice, 0)
-	for i := uint32(0); i < rc.shardCoordinator.NumberOfShards(); i++ {
-		if len(miniBlocks[i].TxHashes) > 0 {
-			finalMiniBlocks = append(finalMiniBlocks, miniBlocks[i])
-		}
+	return nil
+}
+
+func (rc *rewardsCreator) createCommunityRewardTransaction(
+	metaBlock *block.MetaBlock,
+) (*rewardTx.RewardTx, []byte, uint32, error) {
+
+	communityAddress := metaBlock.EpochStart.Economics.CommunityAddress
```

Comment:
community address may be saved as an internal variable to the rewardsCreator - and this check can be done, address validity at the time of constructor.